pkg/operator/ceph/cluster/osd/osd.go

```diff
@@ -341,6 +341,11 @@ func (c *Cluster) startOSDDaemonsOnNode(nodeName string, config *provisionConfig
 			}
 			continue
 		}
+
+		if err = c.deleteDeploymentWithLegacyName(osd.ID); err != nil {
+			logger.Warningf("failed to delete legacy osd deployment. %+v", err)
```

Comment:
if this deletion fails, we will proceed to creating the new OSD deployment and it will not be able to start up right? (due to device busy/in-use). i think that's probably OK instead of bailing out here if deletion of the old deployment failed.